### ISSUE TITLE
Improve game description formatting

### ISSUE THREAD

**Issue by PhrozenByte** [JosefNemec/Playnite#397]
The fact that Playnite expects game descriptions to be HTML caused some minor confusion in the past (#349); even though there's a informative tooltip now, it might still cause confusion - especially because I think that most users don't know what HTML is. Nevertheless is supporting HTML important because some stores use HTML for their game descriptions.

I suggest that the game description field supports Markdown. I guess everyone here on GitHub knows what Markdown is, so I think that I don't have to explain it. The biggest advantage of Markdown is that it is very intuitive (most importantly: paragraphs work as expected) and its simplicity - and, to return to the above "We must support HTML" requirement, it doesn't interfere with HTML in any way and it simply passes HTML through without any change. Just to emphasis this, Markdown parsers don't interpret Markdown inside HTML block elements - thus some `*` somewhere in a game description from Steam won't be parsed as Markdown. Consequently, adding Markdown-support doesn't break backwards compatibility.

I suggest using a [CommonMark](http://commonmark.org/)-compatible parser. There are some C# Markdown parsers out there Playnite could use (e.g. [`lunet-io/markdig`](https://github.com/lunet-io/markdig) or [`Knagis/CommonMark.NET`](https://github.com/Knagis/CommonMark.NET)).

**Comment by JosefNemec** [comment]
I would actually don't mind getting rid of HTML support (yes we would have to convert Steam and GOG descriptions) in favor of something that doesn't render into HTML but FlowDocument instead (like [Markdown.XAML](https://github.com/theunrepentantgeek/Markdown.XAML)).

It would be faster and provide more unified look with the rest of application. Currently HTML descriptions are rendered by [HtmlRenderer.WPF](https://www.nuget.org/packages/HtmlRenderer.WPF), which is basically no longer maintained and has several issues with how the output looks.

**Comment by nathanmerrill** [comment]
+1 to this from me.  Images in descriptions from Epic games use markdown format, which means that Playnite doesn't render them.

**Comment by Deide** [comment]
Can CefSharp not render HTML?

**Comment by Bluscream** [comment]
> 
> 
> Can CefSharp not render HTML?

that would allow the html to do things like run javascript. which you certainly wouldn't want

**Comment by Bluscream** [comment]
If you don't think this would be abused: https://steamdb.info/app/737430/

it is just waiting for steams parser to fuck up in some way

![](https://i.imgur.com/gMWo3af.png)

**Comment by JosefNemec** [comment]
I don't want to use web views because they are slow and memory heavy (quite a lot of people use Playnite on really low power PCs).

@Bluscream  You can disable JS execution in CEF.

**Comment by Deide** [comment]
Makes sense not to have a web view if it's just some text with a couple of images, I agree. Will be a shame to lose the custom styling support, but not a big deal - especially as the SDK updates to accommodate custom elements on a game's view.

**Comment by chocolatechipcats** [comment]
Markdown would be much preferable. 

Hopefully whatever converter you use will be able to handle some of the walls-of-HTML that tend to be downloaded from the Steam store.
